core/src/test/java/org/apache/calcite/plan/volcano/VolcanoRuleCallTest.java

```diff
@@ -0,0 +1,153 @@
+    RelTypeB(RelOptCluster cluster, String label) {
+      super(cluster, label);
+    }
+  }
+
+
+  @Test
+  public void testUnorderedVolcanoDescending() {
+    VolcanoPlanner planner = new VolcanoPlanner();
+    planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
+    planner.addRule(new DummyUnionRule());
+
+    RelOptCluster cluster = newCluster(planner);
+
+        /*
+        test plan:
+        LogicalUnion
+            RelTypeA
+            RelTypeB
+         */
+
+    RelTypeA relA1 =
+        new RelTypeA(
+            cluster,
+            "a1");
+    RelTypeB relB1 =
+        new RelTypeB(
+            cluster,
+            "b1");
+
+    int matchCounter;
+
+
+    planner.ensureRegistered(relA1, null);
+    planner.ensureRegistered(relB1, null);
+
+    LogicalUnion union = LogicalUnion.create(Arrays.asList(relA1, relB1), true);
+
+    // add the original union to planner
+    planner.ensureRegistered(union, null);
+
+    matchCounter = 0;
+    while (planner.ruleQueue.popMatch(VolcanoPlannerPhase.OPTIMIZE) != null) {
+      matchCounter++;
+    }
+    // expected: should match 1 time (incorrect)
+    Assert.assertEquals(1, matchCounter);
+
+  }
+
+
+  @Test
+  public void testUnorderedVolcanoAscending() {
+    VolcanoPlanner planner = new VolcanoPlanner();
+    planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
+    planner.addRule(new DummyUnionRule());
+
+    RelOptCluster cluster = newCluster(planner);
+
+        /*
+        test plan:
+        LogicalUnion
+            RelTypeA
+            RelTypeB
+         */
+
+    RelTypeA relA1 =
+        new RelTypeA(
+            cluster,
+            "a1");
+    RelTypeB relB1 =
+        new RelTypeB(
+            cluster,
+            "b1");
+
+    int matchCounter;
+
+
+    planner.ensureRegistered(relA1, null);
+    planner.ensureRegistered(relB1, null);
+
+    LogicalUnion union = LogicalUnion.create(Arrays.asList(relA1, relB1), true);
+
+    // add the original union to planner
+    planner.ensureRegistered(union, null);
+
+    matchCounter = 0;
+    while (planner.ruleQueue.popMatch(VolcanoPlannerPhase.OPTIMIZE) != null) {
+      matchCounter++;
+    }
+
+        /*
+        test plan:
+        add RelTypeB2 as RelTypeB1's equivalent rel
+        LogicalUnion
+            RelTypeA1
+            (RelTypeB1, RelTypeB2)
+
+        expected: should also trigger the rule
+        the rule specifies unordered - t should trigger the match for any child
+        actual: rule is *not* triggered now
+         */
+    RelTypeB relB2 =
+        new RelTypeB(
+            cluster,
+            "b2");
+    planner.ensureRegistered(relB2, relB1);
+
+    matchCounter = 0;
+    while (planner.ruleQueue.popMatch(VolcanoPlannerPhase.OPTIMIZE) != null) {
+      matchCounter++;
+    }
+    // expected: should match 1 time
+    // because the rule specifies unordered, it should trigger the match for any child
+    Assert.assertEquals(1, matchCounter);
```

Comment:
Please add a message to `assertEquals`. The comment in the code would not be visible in `mvn test`, it would not be visible in `CI`, etc.